### PR TITLE
AB#88301 added question 3 to section 106

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@using HE.Investments.Common.WWW.Components.FormButton
+@using HE.Investment.AHP.Contract.Site
+@using HE.Investment.AHP.WWW.Views.Site
+@model SiteModel
+@{
+    ViewData["Title"] = SitePageTitles.SiteSection106OnlyAffordableHousing;
+}
+<gds-div-grid-column-two-thirds>
+    <gds-back-link href=@(Url.Action("Back", "Site", new { currentPage = SiteWorkflowState.Section106OnlyAffordableHousing, siteId = Model?.Id })!) class="govuk-!-margin-bottom-8"></gds-back-link>
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml:9]: Please use  <vc:ahp-workflow-back-button, also in other views.

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW.Tests/Views/Site/Section106OnlyAffordableHousingTests.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        var errorMessage = "some test error";
+        var modelState = new ModelStateDictionary();
+        modelState.AddModelError(nameof(SiteModel.Section106OnlyAffordableHousing), errorMessage);
+
+        // when
+        var document = await Render<SiteModel>(_viewPath, modelStateDictionary: modelState);
+
+        // then
+        document
+            .HasTitle(SitePageTitles.SiteSection106AffordableHousing)
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW.Tests/Views/Site/Section106OnlyAffordableHousingTests.cs]: Please use HasPageHeader to validate header and caption.

File: src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml (hunk 1 above)
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml:9]: use WorkflowBackButton

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        : base(siteRepository, accountUserContext, logger)
+    {
+    }
+
+    public Task<OperationResult> Handle(ProvideSection106OnlyAffordableHousingCommand request, CancellationToken cancellationToken)
+    {
+        return Perform(
+            site =>
+            {
+                var section106 = site.Section106 ?? new Entities.Section106Entity();
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs:25]: I' am not sure if Section106Entity is an entity. 
What is this entity identifier?
It looks more like VO.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs (hunk 3 above)
Comment (Carq) [reply src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs:25]: Let's discuss this after daily.

File: src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml]: Done.

Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Views/Site/Section106OnlyAffordableHousing.cshtml]: Done

File: src/HE.Investment.AHP.WWW.Tests/Views/Site/Section106OnlyAffordableHousingTests.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW.Tests/Views/Site/Section106OnlyAffordableHousingTests.cs]: Done

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/Site/CommandHandlers/ProvideSection106OnlyAffordableHousingCommandHandler.cs:25]: I will take care of that within question 4 PR

**Pull request merged**